src/services/infra/InfraService.ts

```diff
@@ -565,12 +567,18 @@ export default class InfraService {
       }
 
       if (newLaunchParams.redirectionDomainSource) {
-        const redirectionDomainObject = {
-          source: newLaunchParams.primaryDomainSource,
-          target: REDIRECTION_SERVER_IP,
+        const redirectionDomainObject = REDIRECTION_SERVER_IPS.map((ip) => ({
+          source: newLaunchParams.redirectionDomainSource as string, // checked above
+          target: ip,
           type: RedirectionTypes.A,
-        }
-        message.redirectionDomain = [redirectionDomainObject]
+        })) as [
+          {
+            source: string
+            target: string
+            type: string
+          }
+        ] // type cast is safe since there is at least one redirection domain
```

Comment:
question: a little confused by this, correct me if I'm wrong here. There will be 3 items in the array but the type cast seems to be enforcing it as an array with only a single item?